x/concentrated-liquidity/math_test.go

```diff
@@ -79,27 +75,25 @@ func (suite *KeeperTestSuite) TestLiquidity1() {
 // sqrtPriceB is the larger of sqrtpCur and the nextPrice
 // liquidity0 = amount0 * (sqrtPriceA * sqrtPriceB) / (sqrtPriceB - sqrtPriceA)
 func (suite *KeeperTestSuite) TestLiquidity0() {
-	testCases := []struct {
-		name              string
+	testCases := map[string]struct {
 		currentSqrtP      sdk.Dec
 		sqrtPHigh         sdk.Dec
 		amount0Desired    sdk.Int
 		expectedLiquidity string
 	}{
-		{
-			"happy path",
-			sdk.MustNewDecFromStr("70.710678"),
-			sdk.MustNewDecFromStr("74.161984"),
-			sdk.NewInt(1),
-			"1519.437618821730672389",
-			// https://www.wolframalpha.com/input?i=1+*+%2870.710678+*+74.161984%29+%2F+%2874.161984+-+70.710678%29
+		"happy path": {
+			currentSqrtP:      sdk.MustNewDecFromStr("70.710678118654752440"), // 5000
+			sqrtPHigh:         sdk.MustNewDecFromStr("74.161984870956629487"), // 5500
+			amount0Desired:    sdk.NewInt(1000000),
+			expectedLiquidity: "1519437308.014768571721000000", // TODO: should be 1519437308.014768571720923239
```

Comment:
Convert `liquidity0` to:

```
func liquidity0(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
	if sqrtPriceA.GT(sqrtPriceB) {
		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
	}
	product := sqrtPriceA.Mul(sqrtPriceB)
	diff := sqrtPriceB.Sub(sqrtPriceA)
	return amount.ToDec().Mul(product).Quo(diff)
}
```

That should fix it. `Quo` loses precision. As a result, if we apply it as the last operation, we aren't going to lose as much